clients/index.js

```diff
@@ -300,6 +319,34 @@ function setupChannel(cb) {
     }
 
     self._controlServer.listen(self._controlPort, listenReady.signal);
+
+    function getHostForTChannel() {
+        var host = localIp();
+        var reason = validateHost(host);
+        if (reason !== null) {
+            nextAttempt(host, reason);
+            return;
+        }
+        self.tchannel.listen(self._port, host);
+    }
+
+    function nextAttempt(host, reason) {
+        var delay = self._getHostForTchannelAttemptBackoff * Math.pow(2, getHostAttemptCount);
```

Comment:
Correction: the upper bound with limit = `8` is `4:27.75` (taking the worst case on every random fuzz to be +5%.)